src/Context/TodoContext.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { createContext, useEffect, useState, useRef } from 'react';
+import { Todo } from '../Types/Todo';
+import { useLocalStorage } from '../Hooks/useLocalStorage';
+
+type TodoContextType = {
+  todos: Todo[];
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  addTodo: (title: string) => void;
+  deleteTodo: (id: string) => void;
+  updateTodo: (id: string, title: string) => void;
+  filteredTodos: Todo[];
+  filterBy: string;
+  setFilterBy: React.Dispatch<React.SetStateAction<string>>;
+  deleteCompletedTodos: () => void;
+  headerInputRef: React.RefObject<HTMLInputElement>;
+  focusInput: () => void;
+};
+
+export const TodoContext = createContext<TodoContextType>({
+  todos: [],
+  setTodos: () => {},
+  addTodo: () => {},
+  deleteTodo: () => {},
+  updateTodo: () => {},
+  filteredTodos: [],
+  filterBy: 'all',
+  setFilterBy: () => {},
+  deleteCompletedTodos: () => {},
+  headerInputRef: { current: null },
+  focusInput: () => {},
+});
+
+type TodoProviderProps = {
+  children: React.ReactNode;
+};
+
+export const TodoProvider: React.FC<TodoProviderProps> = ({ children }) => {
+  const [todos, setTodos] = useLocalStorage<Todo[]>('todos', []);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState('all');
+  const headerInputRef = useRef<HTMLInputElement>(null);
+
+  const focusInput = () => {
+    if (headerInputRef.current) {
+      headerInputRef.current.focus();
+    }
+  };
+
+  const addTodo = (title: string) => {
+    const newTodo: Todo = {
+      id: new Date().toString(),
```

Comment:
You can create a function for it, for example, generated or something like that, and use it in all places where you need to generate an id or unique value.